Fix broken SQL indents and run init.sh only once

db260179 opened the pull request:
Hi Gadiener,

Your current master is broken due to incorrect indents when running the sql commands.

This breaks the replication process on startup.

Also the init.sh should be just run once on fresh install and not all the time.

db260179 committed:
Fix init.sh - indents are incorrect

Fix EOSQL indents as breaking the sql commands
Add check for already set MYSQL_GRANT_SLAVE_USER and password from environment variable

db260179 committed:
Only run the init.sh on first install

Copy the init.sh on first install then create a status file to ignore on further restarts

db260179 committed:
Copy init.sh to /usr/local/bin then manage later

Copying init.sh to /usr/local/bin allows us to use this only once on a fresh install

db260179 committed:
Remove docker-compose.dev.yml add readme changes

gadiener commented on Dockerfile:
```diff
@@ -1,6 +1,6 @@
 FROM mariadb:10.3
 
-COPY mariadb/init.sh /docker-entrypoint-initdb.d/
```
Hi @db260179,

the mariadb image already runs the scripts inserted in `/docker-entrypoint-initdb.d/` once, see: https://github.com/docker-library/mariadb/blob/44cfe68144976ce0ec135593ffe15c4bcca0dc6f/10.3/docker-entrypoint.sh#L80. 

So you do not need to manually check in the entrypoint.

db260179 replied:
You are correct, so no need for that change.
I've added a commit to allow the master to confirm to the slave its ready to start. Although you have a 30 second count down it isn't that reliable and relying on the depends_on is not a good method especially on docker swarm

gadiener commented on README.md:
```diff
@@ -109,30 +98,3 @@ MYSQL_SLAVE_SKIP_ERRORS: "all" # (Optional) Default: 'OFF'
 - Permit replication on existing database.
-
-
-## Contributing
```
Hi @db260179,

please leave this section in README file.

db260179 replied:
Are yes, sorry didn't realise I took it out

gadiener commented on mariadb/entrypoint.sh:
```diff
@@ -128,5 +128,10 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
-docker-entrypoint.sh $@
\ No newline at end of file
+# On first install cp the init.sh
+if [ ! -e "/docker-entrypoint-initdb.d/.init" ]; then
```
Hi @db260179,

the `docker-entrypoint-initdb.d` directory isn't a volume, so docker will re-run this at every restart. The `mariadb` image  already runs the scripts inserted in `/docker-entrypoint-initdb.d/` once. So you don't need to check this here.

gadiener commented on README.md:
```diff
@@ -18,7 +15,7 @@ version: '3'
 
 services:
     db-master:
-        image: caffeina/mariadb-replication
```
Hi @db260179,

why force users to build the image when they can use the one already built?

gadiener commented:
Hi @db260179,

thanks for your pull request. I gave you a little review, please answer me as soon as you can so I can merge your pull request.

Thanks for contributing,

Gabriele

db260179 committed:
Remove .init, just check for init.sh

No need to for the .init file as the mariadb container only runs on first run from the /docker-entrypoint-initdb.d
But we need to check if the init.sh exists as we are not assuming that a user might mount the /docker-entrypoint-initdb.d as a volume
The status folder will be used to when master is done and the slave is ready to proceed.

db260179 committed:
Add slave status checks as race conditions can occur

By adding a status function in the init.sh it allows the master to finish its tasks first before the slave can start

db260179 committed:
Revert back the Readme

Unintentional commit change revert back

db260179 committed:
Make the /status folder for slave status

Create a /status folder so that the master can create a status file once finished
This allows the slave to only start once the master is ready.

db260179 reviewed (commented):
Hope this helps and taken in account your comments
David
